ls1intum/Artemis
# [Enhancement] Use separate package.json for cypress

### Motivation and Context


Currently Cypress with all its dependencies is a devDependency in the main `package.json` of Artemis. That means that all the Cypress dependencies are installed even when they are not used (in the Artemis build or when running the client tests).

### Description

This PR adds a separate `package.json` file in the cypress folder and removes the Cypress dependencies from the main `package.json`. I also created a [new buildplan](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TAETSCP) with the changes needed to execute the Cypress tests. I also updated the cypress to the latest version.
I also adjusted the [documentation in confluence](https://confluence.ase.in.tum.de/display/ArTEMiS/Cypress+Tests)

### Steps for Testing


1. Checkout this branch
2. [Configure cypress](https://confluence.ase.in.tum.de/display/ArTEMiS/Cypress+Tests) to run against your target environment
3. Open your favorite shell and navigate to `src/test/cypress`
3. Run `yarn install`
4. Make sure that the cypress dashboard opens when running `yarn cypress:open`
5. Make sure that the tests are still executed when running `yarn cypress:run`

- Code Review
- Manual Tests (executed in buildplan)
